frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -21,20 +26,43 @@ export const ModalProvider: React.FC = ({ children }) => {
   const [isOpen, setOpen] = React.useState(false);
   const [Component, setComponent] = React.useState<ModalComponent>();
   const [componentProps, setComponentProps] = React.useState({});
+  const [componentsMap, setComponentsMap] = React.useState({});
 
   const launchModal = React.useCallback<LaunchModal>(
-    (component, compProps) => {
-      setComponent(() => component);
+    (component, compProps, id = undefined) => {
+      if (id) {
+        setComponentsMap((components) => ({
+          ...components,
+          [id]: { Component: component, props: compProps },
+        }));
+      } else {
+        setComponent(() => component);
+      }
       setComponentProps(compProps);
       setOpen(true);
     },
     [setOpen, setComponent, setComponentProps],
   );
-  const closeModal = React.useCallback<CloseModal>(() => setOpen(false), [setOpen]);
+
+  const closeModal = React.useCallback<CloseModal>(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (_e = null, id = undefined) => {
+      if (id) {
+        setComponentsMap((components) => _.omit(components, id));
+      } else {
+        setOpen(false);
+        setComponent(undefined);
+      }
+    },
+    [setOpen],
+  );
 
   return (
     <ModalContext.Provider value={{ launchModal, closeModal }}>
       {isOpen && !!Component && <Component {...componentProps} closeModal={closeModal} />}
+      {_.map(componentsMap, (c, id) => (
+        <c.Component {...c.props} key={id} closeModal={closeModal} />
```

Comment:
I think close modal needs the id:
```suggestion
        <c.Component {...c.props} key={id} closeModal={(e) => closeModal(e,id)} />
```

Reply:
Thanks @jgbernalp! I refactored a bit based on your suggestion.